app/src/main/kotlin/com/wire/android/ui/home/settings/account/displayname/ChangeDisplayNameScreen.kt

```diff
@@ -57,15 +61,32 @@ import com.wire.android.ui.theme.wireColorScheme
 import com.wire.android.ui.theme.wireDimensions
 import com.wire.android.ui.theme.wireTypography
 
+@RootNavGraph
```

Comment:
shouldn't be @HomeNavGraph ?

Reply:
This `ChangeDisplayNameScreen` is supposed to replace the `HomeScreen` so it's a different navigation entry, that's why it's in the `RootNavGraph`.
Currently we have two NavHosts, the main app's navigation is a `RootNavGraph` and there are all screens that should be open as a separate screen. We got rid of all other nested NavHosts and created one global navigation graph with subgraphs. There's only another NavHost inside `HomeScreen` - it doesn't exactly work like a regular navigation because it's purpose is just to replace screens, so it could be a `AnimatedContent`, but since there is a lot of possible screens to handle there, it's easier to treat it like a navigation within the `HomeScreen` - so it just replaces screens that are supposed to be shown inside the `HomeScreen` so they have access to the bottom tabs, search bar, menu drawer. These screens are in the `HomeNavGraph`- it's the only graph which is not the child of `RootNavGraph`.

- `RootNavGraph` - used in `WireActivity`'s `NavHost` for the global app navigation.
- `HomeNavGraph` - used only in `HomeScreen`'s `NavHost` for switching screens inside `HomeScreen`